# mtDNA

## Steps and ideas

 1. Map mtDNA to the whole genome
	 - bam file

 2. Plot mtDNA segments in R
	 - plot...
	 - in Python
	 - Make sure mdDNA is in the file
		 - `grep mt Daffinis_Female_HiC_FixedScaffolds.fa`
		 - `bioawk -cfastx '$name=="mtDNA" {print ">"$name"\n"$seq}' Daffinis_Female_HiC_FixedScaffolds.fa`
		 - `bioawk -cfastx '$name=="mtDNA" {print ">"$name"\n"$seq}' Daffinis_Female_HiC_FixedScaffolds.fa >Female.mtDNA.fa`
		 - `blastn -query Female.mtDNA.fa -db FemaleHiC2 -outfmt 6 >mtDNAvsFemale.blast`
		 - `grep Chr4 mtDNAvsFemale.blast `
		 - `grep Chr4 mtDNAvsFemale.blast |awk '$9>25000000'`
		 - `grep Chr4 mtDNAvsFemale.blast |awk '$9>25000000' >chr4MtDNA.out`
	 - Downloaded chr4MtDNA.out to KUResearch folder and loaded into R
	 - R script: C:/Users/tdcon/Desktop/KUResearch/PlotmtDNA.R
	 - Check other mtDNA locations
	 - `bioawk -cfastx '$name=="Chr4_MullerB" {print substr($seq,26242008,1979)}' Daffinis_Female_HiC_FixedScaffolds.fa`
	 - 

 3. IVG
	 - Male and Female mtDNA
	 - In Rob/affinisHiC/:
	 - `samtools tview fil_bodyM.fastq.gz.bam Daffinis.Female.HiC.fa`
		 - typed 'g' then the name of what we are looking for: mtDNA_Unscaffolded
	 - Copied these files to my directory and downloaded them to my research folder
		 - Daffinis.Female.HiC.fa
		 - fil_bodyM.fastq.gz.bam
		 - fil_bodyM.fastq.gz.bam.bai
		 - fil_bodyF.fastq.gz.bam
		 - fil_bodyF.fastq.gz.bam.bai
	 - Put those into IVG


